documents/manual_config.txt
//
// Textcube service-wide configurable options via config.php
// ---------------------------------------------------------
//
// This document explains about the available options at config.php
//
<?php
ini_set('display_errors', 'off');
$database['server'] = 'localhost';  // Location of DBMS server
$database['database'] = 'noname';   // Database name
$database['username'] = 'noname';   // user name to estiblish database connection
$database['password'] = 'noname';   // password to establish database connection
$database['prefix'] = 'tt_';        // prefix (useful when you try to install two or more textcubes in one database.
$service['type'] = 'path';          // Service type.
// 'path'   : path-based    e.g.) http://www.example.com/blog1 ,
// 'domain' : domain-based. e.g.) http://blog1.example.com
// If you installed Textcube on http://www.example.com/tc, )
$service['domain'] = 'domain.ext';  // Default service domain. (e.g.) http://www.example.com
$service['path'] = '/path1/path2';  // URI path where textcube is installed (e.g.) /tc
$service['skin'] = 'periwinkle';     // Default blog skin name.
$service['timeout'] = 3600;                 // Session timeout limit (sec.)
$service['disableEolinSuggestion'] = false; // DEPRECATED: Disable auto-suggestion using EOLIN server. 
//Usually it searches local tag by default.
// From Tattertools 1.1
$service['useEncodedURL'] = false; // URL encoding using RFC1738
$serviceURL = ''; // Useful if using other web program under the same domain
// From Textcube 1.5
$service['fancyURL'] = 2;    // fancy-URL control when web server does not support rewrite engine.
// 0 : Do not use fancy-URL,   e.g.) http://www.example.com/index.php?tag
// 1 : Use php direction only. e.g.) http://www.example.com/?tag
// 2 : Use full fancy-URL.     e.g.) http://www.example.com/tag
$service['useNumericURLonRSS'] = false; // Can force permalink to numeric format on RSS output.
$service['useLegacySupport'] = false;   // Uses legacy support (for tattertools plugins) function.
// From Textcube 1.6
$service['flashuploader'] = true;  // Use Flash uploader to upload multiple files.
$service['clipboardpoter'] = true;  // Use Flash clipboard copy to support one-click trackback address copy.
$service['reader'] = true;     // Use Textcube reader. You can set it to false if you do not use Textcube reader, and want to decrease DB load.
$service['pagecache'] = true;  // pagecache function. (changed from disablePageCache option at TC 1.5)
$service['debugmode'] = false; // Textcube debug mode. (for core / plugin debug or optimization)
$service['debug_session_dump'] = false;   // session info debuging.
$service['debug_rewrite_module'] = false; // rewrite handling module info debuging.
$service['allowBlogVisibilitySetting'] = true; // Allow service users to change blog visibility
$service['externalresources'] = false;  // Loads resources from external storage.
$service['resourcepath'] = 'http://example.com/resource';	// Specify the full URI of external resource.
$service['favicon_daily_traffic'] = 10; // Set favicon traffic limitation. default is 10MB.
// From Textcube 1.7
$service['skincache'] = true;        // Use skin pre-fetching.
// Textcube will parse static elements (blog name, title…) only when you change skin.
// From Textcube 1.8
$database['port'] = 3639;            // Database port number
$database['dbms'] = 'MySQL';         // DBMS. (MySQL, MySQLi, PostgreSQL, Cubrid.)
$service['memcached'] = false;       // Using memcache to handle session and cache
$memcached['server'] = 'localhost';  // Where memcache server is.
// From Textcube 1.8.4
$service['requirelogin'] = false;    // Force log-in process to every blogs. (for private blog service)
// From Textcube 1.9.3
$service['useSSL'] = false;         // Use SSL connection. Every http:// will be replaced with https:// or //. When SSL is enabled, every connection header will contain HSTS.
// From Textcube 2.0
$service['jqueryURL'] = '';		// Add URL if you want to use external jquery via CDN. e.g.) Microsoft's CDN: http://ajax.aspnetcdn.com/ajax/jQuery/
$service['lodashURL'] = '';		// Add URL if you want to use external lo-dash via CDN.
?>


library/config.php
<?php
/// Copyright (c) 2004-2009, Needlworks / Tatter Network Foundation
/// All rights reserved. Licensed under the GPL.
/// See the GNU General Public License for more details. (/doc/LICENSE, /doc/COPYRIGHT)

// Define basic signatures.
define('TEXTCUBE_NAME', 'Textcube');
define('TEXTCUBE_VERSION', '1.7.9 : Release Candidate 1');
define('TEXTCUBE_COPYRIGHT', 'Copyright &copy; 2004-2009. Needlworks / Tatter Network Foundation. All rights reserved. Licensed under the GPL.');
define('TEXTCUBE_HOMEPAGE', 'http://www.textcube.org/');
define('TEXTCUBE_RESOURCE_URL', 'http://resources.textcube.org/1.7.9');
define('TEXTCUBE_SYNC_URL', 'http://ping.eolin.com/');
define('CRLF', "\r\n");
define('TAB', "	");
define('INT_MAX',2147483647);
if( strstr( PHP_OS, "WIN") !== false ) {
	define('DS', "\\");
} else {
	define('DS', "/");
}
// Define global variable.
global $database, $service, $blog;

$database['server'] = 'localhost';
$database['database'] = '';
$database['username'] = '';
$database['password'] = '';
$database['prefix'] = '';
$service['timeout'] = 3600;
$service['type'] = 'single';
$service['domain'] = '';
$service['path'] = '';
$service['language'] = 'ko';
$service['timezone'] = 'Asia/Seoul';
$service['encoding'] = 'EUC-KR';
$service['umask'] = 0;
$service['skin'] = 'coolant';
if(defined('__TEXTCUBE_NO_FANCY_URL__')) $service['fancyURL'] = 1;
else $service['fancyURL'] = 2;
$service['useEncodedURL'] = false;
$service['debugmode'] = false;
$service['reader'] = true;
$service['flashclipboardpoter'] = true;
$service['allowBlogVisibilitySetting'] = true;
$service['disableEolinSuggestion'] = false;
$service['effect'] = false;
$service['interface'] = 'detail';	// 'simple' or 'detail'. Default is 'detail'
$service['pagecache'] = true;
$service['codecache'] = false;
$service['skincache'] = true;
$service['externalresources'] = false;
$service['favicon_daily_traffic'] = 10;
$service['flashuploader'] = true;
$service['debug_session_dump'] = false;
$service['debug_rewrite_module'] = false;
$service['useNumericURLonRSS'] = false;
// Map port setting.
if (@is_numeric($_SERVER['SERVER_PORT']) && ($_SERVER['SERVER_PORT'] != 80) && ($_SERVER['SERVER_PORT'] != 443))
	$service['port'] = $_SERVER['SERVER_PORT'];

// Include installation configuration.
$service['session_cookie_path'] = '/';
if(!defined('__TEXTCUBE_SETUP__')) @include ROOT . '/config.php';

if($service['debugmode'] == true) requireComponent("Needlworks.Debug.".$database['dbms']);

// Set resource path.
if($service['externalresources']) {
	if(isset($service['resourceURL']) && !empty($service['resourceURL'])) 
		$service['resourcepath'] = $service['resourceURL'];
	else 
		$service['resourcepath'] = TEXTCUBE_RESOURCE_URL;
} else {
	$service['resourcepath'] = empty($serviceURL) ? $service['path']:$serviceURL;
}

if(!empty($service['domain']) && strstr( $_SERVER['HTTP_HOST'], $service['domain'] ) ) {
	$service['session_cookie_domain'] = $service['domain'];
} else {
	$service['session_cookie_domain'] = $_SERVER['HTTP_HOST'];
}
if ($service['session_cookie_domain'] == 'localhost') 
	$service['session_cookie_domain'] = null; 

// Basic POST/GET variable validation.
$valid = true;
if(isset($IV)) $valid = $valid && Validator::validate($IV);

// Basic SERVER variable validation.
$basicIV = array(
	'SCRIPT_NAME' => array('string'),
	'REQUEST_URI' => array('string'),
	'REDIRECT_URL' => array('string', 'mandatory' => false)
);
$valid = $valid && Validator::validateArray($_SERVER, $basicIV);
if(isset($accessInfo)) {
	$basicIV = array(
		'fullpath' => array('string'),
		'input'    => array('string'),
		'position' => array('string'),
		'root'     => array('string'),
		'input'    => array('string', 'mandatory' => false)
	);
	$accessInfo['fullpath'] = urldecode($accessInfo['fullpath']);
	$valid = $valid && Validator::validateArray($accessInfo, $basicIV);
}
if(!$valid) {
	header('HTTP/1.1 404 Not Found');
	exit;
}
?>
